teachertool/src/App.tsx

```diff
@@ -35,28 +38,37 @@ export const App = () => {
                 await loadValidatorPlansAsync();
                 await tryLoadLastActiveChecklistAsync();
 
-                // Test notification
-                showToast({
-                    ...makeToast("success", "🎓", 2000),
-                    hideIcon: true,
-                    hideDismissBtn: true,
-                    className: "app-large-toast",
-                });
-
                 setInited(true);
                 logDebug("App initialized");
             });
         }
     }, [ready, inited]);
 
-    return !inited ? (
+    useEffect(() => {
+        async function checkAuth() {
+            // On mount, check if user is signed in
+            try {
+                await authClient.authCheckAsync()
+            } catch (e) {
+                // Log error but continue
+                // Don't include actual error in error log in case PII
+                logError(ErrorCode.authCheckFailed);
+                logDebug("Auth check failed details", e);
+            }
+            setAuthCheckComplete(true)
+        }
+        checkAuth();
```

Comment:
Should this be awaited?

Reply:
useEffects can't be async, so we can't await here, but the important thing is just to kick it off.